### PR TITLE
[MXNET][NEURON][BUILD] - 1.8 use latest neuron rel

### PR DESCRIPTION
**Pull request by aws-vrnatham**
Signed-off-by: Venky Natham <vrnatham@amazon.com>

*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Modify the mxnet18 to use the latest neuron release where runtime is removed

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron"/"graviton">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by saimidu — DISMISSED** [review]
Approved. Please respond to the questions in the review and revert the temporary config changes when ready, for the final approval+merge.

File: mxnet/inference/docker/1.8/py3/sdk1.16.0/Dockerfile.neuron [hunk 1]
```diff
@@ -116,7 +115,14 @@ RUN useradd -m model-server \
...
  && chown -R model-server /home/model-server
 
-COPY mms-entrypoint.py /usr/local/bin/dockerd-entrypoint.py
+COPY neuron-entrypoint.py /usr/local/bin/dockerd-entrypoint.py
+COPY neuron-start.sh /usr/local/bin/neuron-start.sh
+COPY modelserver-neuron.sh /usr/local/bin/entrypoint.sh
+
+RUN chmod +x /usr/local/bin/dockerd-entrypoint.py
+RUN chmod +x /usr/local/bin/neuron-start.sh
+RUN chmod +x /usr/local/bin/entrypoint.sh
```
[the remark this answers is not in the document]
Comment (aws-vrnatham) [reply mxnet/inference/docker/1.8/py3/sdk1.16.0/Dockerfile.neuron]: Done

File: mxnet/inference/docker/artifacts/neuron-entrypoint.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import shlex
+import subprocess
+import sys
+import os.path
+
+neuron_cmd = "/usr/local/bin/neuron-start.sh"
```
[the remark this answers is not in the document]
Comment (aws-vrnatham) [reply mxnet/inference/docker/artifacts/neuron-entrypoint.py]: If they do that then they just lose the ability to collect metrics in the cloudwatch etc.. Functionally the container will still be able to do ineference etc.

File: test/dlc_tests/sanity/test_pre_release.py [hunk 3]
```diff
@@ -228,7 +228,15 @@ def test_framework_and_neuron_sdk_version(neuron):
         container_name, ctx, f"import {tested_framework}; print({tested_framework}.__version__)", executable="python"
     )
 
-    assert neuron_tag_framework_version == output.stdout.strip()
+    if tested_framework == "mxnet":
+        # TODO -For neuron the mx_neuron module does not support the __version__ yet and we
+        # can get the version of only the base mxnet model. The base mxnet model just
+        # has framework version and does not have the neuron semantic version yet. Till
+        # the mx_neuron supports __version__ do the minimal check and not exact match
```
[the remark this answers is not in the document]
Comment (aws-vrnatham) [reply test/dlc_tests/sanity/test_pre_release.py:235]: In the next neuron sdk release.

**Review by saimidu — DISMISSED** [review]
Approved. Please revert temporary config file changes in order to make the PR merge-ready.

**Review by saimidu — APPROVED** [review]

**Pull request merged**